Libraries/Opc.Ua.Client/Session/SessionConfiguration.cs

```diff
@@ -44,101 +52,142 @@ namespace Opc.Ua.Client
     [KnownType(typeof(X509IdentityToken))]
     [KnownType(typeof(IssuedIdentityToken))]
     [KnownType(typeof(UserIdentity))]
-    public class SessionConfiguration
+    public record class SessionOptions
     {
         /// <summary>
-        /// Creates a session configuration
+        /// The session name used by the client.
+        /// </summary>
+        [DataMember(IsRequired = true, Order = 20)]
+        public string? SessionName { get; set; }
+
+        /// <summary>
+        /// The identity used to create the session.
+        /// </summary>
+        [DataMember(IsRequired = true, Order = 50)]
+        public IUserIdentity? Identity { get; set; }
+
+        /// <summary>
+        /// The configured endpoint for the secure channel.
+        /// </summary>
+        [DataMember(IsRequired = true, Order = 60)]
+        public ConfiguredEndpoint? ConfiguredEndpoint { get; set; }
+
+        /// <summary>
+        /// If the client is configured to check the certificate domain.
+        /// </summary>
+        [DataMember(IsRequired = false, Order = 70)]
```

Comment:
If not required, the type should be nullable?

Reply:
Should be the default which is false.  DataMember and contract just enforce serialization and barf if the value is missing in the xml.